templates/shared_services/nexus-cert/terraform/appgateway.tf

```diff
@@ -0,0 +1,171 @@
+resource "azurerm_public_ip" "appgwpip" {
+  name                = "pip-nexus-${var.tre_id}"
+  resource_group_name = local.core_resource_group_name
+  location            = var.location
+  allocation_method   = "Static"
+  sku                 = "Standard"
+  domain_name_label   = "nexus-${var.tre_id}"
```

Comment:
If you made the whole domain_name_label a var, you are nearly 90% complete on making this re-usuable

Reply:
As is, its a 1-1 relationship, in the future this could achieve the fully dynamic shared service 
that uses a  dynamic tf file to achieve the goal. But as an is this is needed  
Asia k was focusing on getting the Nexus/Docker solution configured for the current needs and ammended dependant on future needs while the wider issues is addressed